### ISSUE TITLE
Add icon support

### ISSUE THREAD

**Issue by filisko** [r0x0r/pywebview#185]
Hello,
Is it possible to add window icon support? I'm using this in Linux and I can't find a way to do that.

**Comment by r0x0r** [comment]
If you freeze your app, you might want to look the icon support of your
favorite freezing software. Other than that there is no icon support.

**Comment by filisko** [comment]
Yes, the thing is that I don't want to freeze it :smile:.

Ok, thank you.

**Comment by r0x0r** [comment]
I have nothing against adding icon support, say `create_window(icon='path to the icon file or something like that')`, provided there is a neat cross-platform solution or at least Linux+OSX. The problem is that all the platforms use different icon file formats, so this issue has to be solved first.

**Comment by r0x0r** [comment]
I am not going to work on this. Feel free to reopen if you wish to give this a go.

**Comment by rukameshkumar-tal** [comment]
I have created a GUI application using pywebview in python and want to change application icon shown in menu bar, By default it is shown as python icon in windows just want to check is there any way to change default icon to custom icon?

**Comment by r0x0r** [comment]
The current workaround is to freeze your application and supply your icon via the freezer. It will appear just fine in the application. Pyinstaller is a good one. 
I see little value in implementing an icon support when running as a script, so I won't work on it. If somebody wants to implement it, then by all means

**Comment by rukameshkumar-tal** [comment]
@r0x0r Thanks for your suggestion One more issue that I am currently facing in macOS I have created a custom button to close the current window for that I have used in same way how it is defined here https://stackoverflow.com/questions/67208598/close-pywebview-window-with-html-button-click but it close the window but does not close the python process same thing is working fine in windows but not in macOS so could you please help on same

**Comment by r0x0r** [comment]
Can you use sys.exit?

**Comment by rukameshkumar-tal** [comment]
I have used this with  inclosing as well quit method but due to this
application got hanged so not sure what is the right way to use same?

**Comment by r0x0r** [comment]
See this discussion
https://stackoverflow.com/questions/1489669/how-to-exit-the-entire-application-from-a-python-thread

TLDR: you may use os._exit(0) for quitting your app

**Comment by shivjeetbhullar** [comment]
In the PyWebView framework if you are using the QT web engine as GUI then you have to add a window icon manually in the site packages **"\Python39\Lib\site-packages\webview\platforms\qt.py"** file. Simply paste 2 lines into class BrowserView Decorator __init__().
<pre>
   class BrowserView(QMainWindow):
      def __init__(self, window):
        from PyQt5 import QtGui
        self.setWindowIcon(QtGui.QIcon('icon.ico'))
</pre>

**Comment by DaichiFujiwara** [comment]
For windows platform, we can set up a custom app icon by modifying "\path\to\site-packages\webview\platforms\winforms.py" 
 implementation.
From line 235, "on_activated" method codes (in BrowserForm class) gets started.
I modified as follows and successfully showed up custom app icon.

`
def on_activated(self, *_):
            if not self.pywebview_window.focus:
                windll.user32.SetWindowLongW(self.Handle.ToInt32(), -20, windll.user32.GetWindowLongW(self.Handle.ToInt32(), -20) | 0x8000000)
            elif self.browser:
                self.browser.web_view.Focus()

            if is_cef and self.pywebview_window.focus:
                CEF.focus(self.uid)
            
            # [add] custom icon
            icon = Icon("/path/to/icon/image")

            # [add] custom icon
            form = WinForms.Application.OpenForms[0]
            form.Icon = icon
`

**Comment by SquashyHydra** [comment]
> For windows platform, we can set up a custom app icon by modifying "\path\to\site-packages\webview\platforms\winforms.py" implementation. From line 235, "on_activated" method codes (in BrowserForm class) gets started. I modified as follows and successfully showed up custom app icon.
> 
> ```python
> def on_activated(self, *_):
>         if not self.pywebview_window.focus:
>             windll.user32.SetWindowLongW(self.Handle.ToInt32(), -20, windll.user32.GetWindowLongW(self.Handle.ToInt32(), -20) | 0x8000000)
>         elif self.browser:
>             self.browser.web_view.Focus()
> 
>         if is_cef and self.pywebview_window.focus:
>             CEF.focus(self.uid)
>             
>         # [add] custom icon
>         icon = Icon("/path/to/icon/image")
> 
>         # [add] custom icon
>         form = WinForms.Application.OpenForms[0]
>         form.Icon = icon
> ```

I got something sort of working for myself for the windows platform if you want something more like this
`    window = webview.create_window("Title", "http://127.0.0.1:5000", icon=icon_path)`
```
    window = webview.create_window("Title", "http://127.0.0.1:5000")
    window.confirm_close = True
    window.icon = icon_path
```
in packages/webview/window.py go to line 106 after the Window class is initialized `class Window:`
and add
`icon: str | None = None`
```
        localization: Mapping[str, str] | None = None,
        http_port: int | None = None,
        server: type[http.ServerType] | None = None,
        server_args: http.ServerArgs = {},
        screen: Screen = None,
        icon: str | None = None
    ) -> None:
```
Bellow that on line 137 
add this line
`        self.icon = icon`
```
        self.transparent = transparent
        self.zoomable = zoomable
        self.draggable = draggable
        self.localization_override = localization
        self.vibrancy = vibrancy
        self.screen = screen
        self.icon = icon

```
save and close window.py and open __init__.py navigate to line 246 under the `def create_window( `
and add this line
`    icon: str | None = None,`
```
    zoomable: bool = False,
    draggable: bool = False,
    vibrancy: bool = False,
    localization: Mapping[str, str] | None = None,
    server: type[http.ServerType] = http.BottleServer,
    http_port: int | None = None,
    server_args: http.ServerArgs = {},
    icon: str | None = None,
) -> Window:
```
navigate down to line 317 under `window = Window(`
add this line
`        icon=icon,`
```
        vibrancy,
        localization,
        server=server,
        http_port=http_port,
        server_args=server_args,
        screen=screen,
        icon=icon,
    )

    windows.append(window)
```
save and close __init__.py and navigate to packages/webview/platform/winforms.py and go to line 160 `class BrowserForm(WinForms.Form):` and add a custome icon variable
`            self.Custome_Icon = window.icon`
```
        def __init__(self, window, cache_dir):
            super().__init__()
            self.uid = window.uid
            self.pywebview_window = window
            self.real_url = None
            self.Text = window.title
            self.Custome_Icon = window.icon
            self.Size = Size(window.initial_width, window.initial_height)
            self.MinimumSize = Size(window.min_size[0], window.min_size[1])
```
scroll down to line 265 bellow `def on_activated(self, *_):`
add these lines to the function
```
            elif self.browser:
                self.browser.web_view.Focus()
                
            if is_cef and self.pywebview_window.focus:
                CEF.focus(self.uid)
            
            icon = Icon(self.Custome_Icon)

            form = WinForms.Application.OpenForms[0]
            form.Icon = icon
```
```
        def on_activated(self, *_):
            if not self.pywebview_window.focus:
                windll.user32.SetWindowLongW(self.Handle.ToInt32(), -20, windll.user32.GetWindowLongW(self.Handle.ToInt32(), -20) | 0x8000000)
            elif self.browser:
                self.browser.web_view.Focus()
                
            if is_cef and self.pywebview_window.focus:
                CEF.focus(self.uid)
            
            icon = Icon(self.Custome_Icon)

            form = WinForms.Application.OpenForms[0]
            form.Icon = icon
```
